### PR TITLE
CRM-20008 fix pending flow.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
This adds a test written by Peter Hartmann for repeattransaction with pending status.

I altered is a little, working on the assumption the flow should be
- create pending, leave status unchanged
- complete pending, updte status & advance end date

---

 * [CRM-20008: repeattrasaction should create pending membership when contribtuion_status_id is pending](https://issues.civicrm.org/jira/browse/CRM-20008)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@h-c-c do you want to check this? I looked at your fix #9827 - but it seemed to me cleaner to deal with it a bit deeper.

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -4548,43 +4548,52 @@ public static function completeOrder(&$input, &$ids, $objects, $transaction, $re
             }
             $dao->free();
 
-            $membershipParams['num_terms'] = $contribution->getNumTermsByContributionAndMembershipType(
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: change is just indentation - the if -else block

File: CRM/Contribute/BAO/Contribution.php [hunk 2]
```diff
@@ -4548,43 +4548,52 @@ public static function completeOrder(&$input, &$ids, $objects, $transaction, $re
...
-              $dates = CRM_Member_BAO_MembershipType::getRenewalDatesForMembershipType($membershipParams['id'],
-                $changeDate, NULL, $membershipParams['num_terms']
-              );
-
-              $dates['join_date'] = $currentMembership['join_date'];
+            if (CRM_Core_PseudoConstant::getLabel('CRM_Contribute_BAO_Contribution', 'contribution_status_id', CRM_Utils_Array::value('contribution_status_id', $input)) === 'Pending') {
+              $membershipParams['num_terms'] = 0;
             }
+            else {
+              $membershipParams['num_terms'] = $contribution->getNumTermsByContributionAndMembershipType(
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: but the else block is just formatting

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
@eileenmcnaughton just re-testing to check git scan works and everything is hunky dory after your merge of the previous pull

File: CRM/Contribute/BAO/Contribution.php [hunk 3]
```diff
@@ -4548,43 +4548,52 @@ public static function completeOrder(&$input, &$ids, $objects, $transaction, $re
...
-              CRM_Member_BAO_Membership::fixMembershipStatusBeforeRenew($currentMembership, $changeDate);
-
-              // @todo - we should pass membership_type_id instead of null here but not
-              // adding as not sure of testing
-              $dates = CRM_Member_BAO_MembershipType::getRenewalDatesForMembershipType($membershipParams['id'],
-                $changeDate, NULL, $membershipParams['num_terms']
-              );
-
-              $dates['join_date'] = $currentMembership['join_date'];
+            if (CRM_Core_PseudoConstant::getLabel('CRM_Contribute_BAO_Contribution', 'contribution_status_id', CRM_Utils_Array::value('contribution_status_id', $input)) === 'Pending') {
```
Comment (h-c-c) [inline CRM/Contribute/BAO/Contribution.php]: Nice work!!  Yeah, makes sense to add coverage for the whole process.

**Comment by eileenmcnaughton** [comment]
@h-c-c Can you confirm if you have tested this / read the code & feel it is good to merge. I need to get some review before I can merge.  I feel quite good about the test case adding coverage for the scenario

Note that most of the change is moving code

**Comment by seamuslee001** [comment]
This looks correct to me no num_terms if we are pending otherwise check as per previous

**Comment by eileenmcnaughton** [comment]
Thanks @seamuslee001 - merging

**Pull request merged**